gdal/src/main/scala/geotrellis/contrib/vlm/gdal/GDALDataPath.scala

```diff
@@ -0,0 +1,30 @@
+package geotrellis.contrib.vlm.gdal
+
+import geotrellis.contrib.vlm.DataPath
+
+import java.net.MalformedURLException
+
+
+case class GDALDataPath(val path: String) extends DataPath {
+  protected def servicePrefix: String = "gdal+"
+
+  private lazy val strippedPath: String =
+    if (path.startsWith(servicePrefix))
+      path.splitAt(servicePrefix.size)._2
+    else
+      path
+
+  private lazy val vsiPath: VSIPath = VSIPath(strippedPath)
+
+  def targetPath: String =
```

Comment:
I don't think this is part of public interface either. The parameter `path` can be used in `toString` and each individual subclass can have a field that represented the parsed/transformed value like `.vsiPath` for GDAL and `.catalogUri` and `.layerName` for GeoTrellis catalog etc.